databases/business-sync/fragment-shoebox-snippet.js

```diff
@@ -0,0 +1,40 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    return createBusinessEntityRegex('shoeboxSnippet\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox snippet (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    source: {
+      // The source of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    kashooEntityType: {
+      // Type of corresponding Books entity
+      type: 'string',
+      required: false,
+      mustNotBeEmpty: true
+    },
+    kashooId: {
+      // The ID of the corresponding Books entity
+      type: 'integer',
+      required: false,
+      minimumValue: 1
+    },
+    data: {
+      // Raw data of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    }
```

Comment:
Yeah, I can't say I fully on board with a `processed` flag.  You make some good points, but is a simple flag sufficient?  Will there only be a single time we process any given item?  It might be good to track artifacts of processing, but maybe be more flexible as an array of associated processed document IDs (and maybe the processed timestamp as well?  although that would certainly also be a field in the 'processed' document)